pages/directory/index.tsx

```diff
@@ -8,6 +9,56 @@ import createAxiosInstance from 'client/lib/axios';
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const Directory: NextPage<{ users: any[]; error: string }> = props => {
   const { users, error } = props;
+  const lastCardRef = React.useRef(null);
+  const [userData, setUserData] = React.useState(users);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [totalPages, setTotalPages] = React.useState(999);
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  async function fetchUsers(page: number) {
+    try {
+      const response = await axios.get(`/api/users?page=${page}`);
+      return {
+        page: response.data.page,
+        total: response.data.totalPages,
+        users: response.data.users,
+      };
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  React.useEffect(() => {
+    const intersectionOptions = {
+      // root default = document's viewport
+      // in the future offset should be tied to the card height of the last row
+      rootMargin: '300px',
+      threshold: 0,
+    };
```

Comment:
300px is a temp value, In the future, the offset should be based on the height of one row of user profiles